### ISSUE TITLE
stdint compat for windows

### ISSUE THREAD

**Issue by jbrockmendel** [cython/cython#2570]
I have a few cython modules that currently get integer type definitions from numpy using `from numpy cimport int64_t, int32_t`.  Related to #2569, I would prefer to remove the numpy dependency and just use `from libc.stdint cimport int64_t, int32_t`.  Naturally this works on OSX/Linux but fails on Windows.

pandas has a shim for stdint on Windows: https://github.com/pandas-dev/pandas/blob/master/pandas/_libs/src/headers/stdint.h

Would it make sense for this to be done upstream in cython?

**Comment by scoder** [comment]
Now that we support verbatim C code in .pxd files, I'm all for it. This has annoyed me way too often.

**Comment by gabrieldemarmiesse** [comment]
We'll finally be able to use `intptr_t` in the docs!

**Comment by Yoshi325** [comment]
I'm willing to give this a go.

Does this require #2183 ?

**Comment by scoder** [comment]
I don't think it needs #2183. The C code should still be injected early enough.

**Comment by jbrockmendel** [comment]
Removing the windows shim in pandas-dev/pandas#46437 didn't break the CI.  Does that suggest something here is actionable?

**Comment by jakirkham** [comment]
Historically Python versions and binaries that used them had to compile with [particular versions of MSVC]( https://wiki.python.org/moin/WindowsCompilers ), which was determined by what MSVC was available around the time of the Python release and needed to be tightly constrained due to API/ABI compatibility. With [MSVC 2015 and UCRT]( https://docs.microsoft.com/en-us/cpp/porting/upgrade-your-code-to-the-universal-crt?view=msvc-170 ), which lined up with the Python 3.5 release, this became considerably more relaxed. So Python versions 3.5+ have been able to use any MSVC 2015+ (IOW VC++ 14.x) version. As older Python versions were dropped from support fewer old MSVC versions have needed support. The last of these was Python 2.7, which used MSVC 2008.

Beginning in MSVC 2013 (IOW VC++ 12) support for various [C99 features were added including `stdint.h`]( https://devblogs.microsoft.com/cppblog/whats-new-for-visual-c-developers-in-vs2013-preview/#c99-libraries ). However Python never supported this version and instead jumped to MSVC 2015, which included this functionality.

So libraries that require Python 3.5 or newer, can just use MSVC without any changes needed here. Workarounds that vendored `stdint.h` can be dropped in these cases as well.
